synthesizer/process/src/lib.rs

```diff
@@ -124,17 +124,22 @@ impl<N: Network> Process<N> {
         let credits_program_id = ProgramID::<N>::from_str("credits.aleo")?;
         // If the program is not 'credits.aleo', compute the program stack, and add it to the process.
         if program.id() != &credits_program_id {
-            self.add_stack(Stack::new(self, program)?);
+            self.add_stack(Stack::new(self, program)?)?;
         }
         Ok(())
     }
 
     /// Adds a new stack to the process.
     /// If you intend to `execute` the program, use `deploy` and `finalize_deployment` instead.
     #[inline]
-    pub fn add_stack(&mut self, stack: Stack<N>) {
-        // Add the stack to the process.
-        self.stacks.insert(*stack.program_id(), Arc::new(stack));
+    pub fn add_stack(&mut self, stack: Stack<N>) -> Result<()> {
```

Comment:
```suggestion
    pub fn add_stack(&mut self, stack: Stack<N>) {
```
the body is infallible